### ISSUE TITLE
support cgroup mode

### ISSUE THREAD

**Issue by maximelavocat** [terraform-google-modules/terraform-google-kubernetes-engine#1838]
### TL;DR

Hi, would it be possible to have [cgroup mode](https://github.com/hashicorp/terraform-provider-google/pull/16103) supported in the next version of this module?

### Terraform Resources

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#cgroup_mode
https://github.com/hashicorp/terraform-provider-google/blob/HEAD/CHANGELOG.md#510-Oct-9-2023

### Additional information

I know this mode was introduced in v5.1 of the provider while the current version of the module (v29) is based on v5.0 on the provider.

Not sure how the development of this module is working, will it be embedded "automatically" in the next version (v30) since it will be based at least on v5.1 of the provider? Or each new feature need to be selected to be in the future version of this module?
